src/libxtp/tools/mol2orb.cc

```diff
@@ -0,0 +1,272 @@
+#include "mol2orb.h"
+
+// Third party includes
+#include <boost/algorithm/string.hpp>
+
+// Local VOTCA includes
+#include "votca/xtp/ecpaobasis.h"
+#include "votca/xtp/orbitals.h"
+#include <votca/tools/constants.h>
+
+// Local VOTCA includes
+#include "votca/xtp/logger.h"
+
+namespace votca {
+namespace xtp {
+
+void Mol2Orb::Initialize(const tools::Property& user_options) {
+  tools::Property options =
+      LoadDefaultsAndUpdateWithUserOptions("xtp", user_options);
+
+  _job_name = options.ifExistsReturnElseReturnDefault<std::string>("job_name",
+                                                                   _job_name);
+  _moldenfile = _job_name + ".molden.input";
+  _orbfile = _job_name + ".orb";
+  _xyzfile = _job_name + ".xyz";
+
+  _basisset_name = options.get(".basisset").as<std::string>();
+  _aux_basisset_name = options.get(".auxbasisset").as<std::string>();
+}
+
+void Mol2Orb::setupBasis(Orbitals& orbitals) {
+  // We only need the basis set for checking correctness
+  BasisSet bs;
+  bs.Load(_basisset_name);
+
+  // fill DFT AO basis by going through all atoms
+  _basis.Fill(bs, orbitals.QMAtoms());
+  orbitals.setDFTbasisName(_basisset_name);
+  XTP_LOG(Log::error, _log)
+      << TimeStamp() << " Basisset size: " << _basis.AOBasisSize()
+      << std::flush;
+  orbitals.setBasisSetSize(_basis.AOBasisSize());
+  orbitals.setAuxbasisName(_aux_basisset_name);
+}
```

Comment:
The name is not Good, because what you are actually doing is adding the basisset information to orbitals.